### PR TITLE
Add hungarian keyboard layout support

### PR DESCRIPTION
**Pull request by Picsiri**
Title says everything

### CODE REVIEW

File: src/KeyboardLayout_hu_HU.cpp
Comment (edgar-bonet) [inline src/KeyboardLayout_hu_HU.cpp:108]: I believe this one is wrong. According to [kbdlayout.info][], this key combination yields a [dead][] circumflex accent (can you confirm this?), whereas this position in the table should map to the ASCII caret (code point U+005E), which is not a dead key, but a character in itself.

If this is indeed the case, this position should be marked as `0x00`. C.f. this explanation on pull request #53:

> Characters requiring hitting multiple keys _consecutively_ (like dead circumflex + space → `^` on some layouts) are not supported for the same reason. [namely: this would require growing the ASCII map to 16&nbsp;bits per cell, which would cost 128&nbsp;extra bytes of flash.]

[kbdlayout.info]: http://kbdlayout.info/KBDHU/
[dead]: https://en.wikipedia.org/wiki/Dead_key

File: src/Keyboard_hu_HU.h
Comment (edgar-bonet) [inline src/Keyboard_hu_HU.h:49]: You should add 136 to the scan codes here, e.g.

```c++
#define KEY_O_SLASH             (136+0x2e)
```

This is for reasons pertaining to the internal encoding used by the library: codes starting at 136 are interpreted as (shifted) raw scan codes. In contrast, `0x2e` (the same as `'.'`) is interpreted as the ASCII full stop.

I suggest also renaming these macros:

* `KEY_*_SLASH` → `KEY_*_ACUTE`. This is in order to be consistent with the Unicode names of the characters (“LATIN SMALL/CAPITAL LETTER A/E/I/O/U WITH ACUTE”), and with the names already used in the library.

* `KEY_*_DOUBLE_SLASH` → `KEY_*_DOUBLE_ACUTE`, for consistency with Unicode (there is no double acute in the library yet).

* `KEY_*_TWO_DOTS` → `KEY_*_UMLAUT` (for consistency with the de\_DE and sv\_SE layouts in this library) or `KEY_*_DIARESIS` (for consistency with Unicode).

**Review by edgar-bonet — CHANGES_REQUESTED** [review]
Hello, thanks for this pull request! It's nice to see people contributing to the internationalization of this library. I am not a maintainer, but I have a few suggestions below.

**Comment by edgar-bonet** [comment]
@facchinm

> Would you fancy becoming it 🙂 ?

Well, yes! I would be honored. Really.

Note that I have only one Arduino on which I can test this library: an Arduino Micro. I thus cannot check cross-architecture compatibility. Is this a problem?

> I'd really love to give you the freedom to apply changes and speedup the whole process 😉

What would be the recommended way of handling the pull request then? Wait for the author to amend or respond to the feedback? Or push my recommended changes on top of his commit and then merge?

File: src/KeyboardLayout_hu_HU.cpp
Comment (Picsiri) [reply src/KeyboardLayout_hu_HU.cpp:108]: Oh my, I barely understand 10% of what you said, but I think you mean that '^' requires me to press AltGr + 3 and than another button to be typed. Like a space (or any other).

Other chars you mentioned do not require this extra button press.

Should I remove the '^' part? Also I checked and unfortanetely I did not test for that char. The test code:
````
const char* specialChars[] = {
    "_ underscore",
    "? question mark",
    "! exclamation mark",
    "# hashtag",
    "% percent",
    "* asterisk",
    "$ dollar sign",
    "@ at symbol",
    "& ampersand",
    "| vertical bar",
    "\\ backslash",
    "/ forward slash",
    "\" double quote",
    "' single quote",
    "` backtick",
    "~ tilde"
    "; semicolon",
    ": colon",
    ", comma",
    ". dot",
    "- hyphen",
    "+ plus",
    "= equals",
    "() parentheses",
    "[] brackets",
    "{} braces",
    "<> angle brackets",
    "§ section sign",
  };
````ˇ

I made it in a hurry... It's a bit sloppy :/

File: src/KeyboardLayout_hu_HU.cpp
Comment (Picsiri) [reply src/KeyboardLayout_hu_HU.cpp:144]: Fixed it, also, why is this important?

File: src/Keyboard_hu_HU.h
Comment (Picsiri) [reply src/Keyboard_hu_HU.h:4]: Ooops...

File: src/Keyboard_hu_HU.h
Comment (Picsiri) [reply src/Keyboard_hu_HU.h:49]: I'm using copilot. It suggested ACUTE but I looked it up on google and I saw a boy with a hurt finger and I was like no thank you :D Looks like copilot was right after all...

Also thank you for typing out the fix so accurately.

**Comment by Picsiri** [comment]
> What would be the recommended way of handling the pull request then? Wait for the author to amend or respond to the feedback? Or push my recommended changes on top of his commit and then merge?

Man, go ahead! :D

File: src/KeyboardLayout_hu_HU.cpp
Comment (edgar-bonet) [reply src/KeyboardLayout_hu_HU.cpp:108]: > Should I remove the '^' part?

I think you can leave it after all, as on some variations of the Hungarian layout AltGr+3 gives `^` without requiring an extra space.

File: src/KeyboardLayout_hu_HU.cpp
Comment (edgar-bonet) [reply src/KeyboardLayout_hu_HU.cpp]: It is not essential, but just keeps things tidier:

* The text format is consistent across the code base.

* `git diff` and `git show` stop warning about “No newline at end of file”.

* If someone later edits those files with a Unixish editor, we won't get spurious changes in the diff because of the missing newline being added by the editor.

**Comment by edgar-bonet** [comment]
Merged. @Picsiri: thanks for your contribution!

**Pull request merged**
